### PR TITLE
Crdt entity fuzz tests

### PR DESCRIPTION
**Pull request by SHeimlich**
Add a fuzz test that checks that the changes are empty for:
```
val changes = EntityA.merge(entityA.data)
```

### CODE REVIEW

File: javatests/arcs/core/crdt/CrdtEntityFuzzTest.kt [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      GetInt(it, 0, 3)
+    )
+
+    val collections = MapOf(
+      ChooseFromList(it, listOf("koalas", "kangaroos", "penguins", "emus")),
+      SetOf(
+        ReferencableGenerator(
+          ChooseFromList(
+            it,
+            listOf("animals, mammals, reptiles", "shelleds", "pets", "birds", "fish")
```
Comment (shans) [inline javatests/arcs/core/crdt/CrdtEntityFuzzTest.kt:35]: should animals, mammals, reptiles be a single option, or 3 separate options? A comment might be good if they're meant to be a single option.

File: javatests/arcs/core/crdt/CrdtEntityFuzzTest.kt [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      Value(100),
+      Value(100)
+    )
+
+    val entity1 = CrdtEntityGenerator(
+      versionMap,
+      rawEntity
+    )()
+
+    val changes = entity1.merge(entity1.data)
```
Comment (shans) [inline javatests/arcs/core/crdt/CrdtEntityFuzzTest.kt:56]: express this as a separate invariant, in javatests/arcs/core/crdt/CrdtEntityInvariants.kt (actually, you could also make it generic as this is a property of all CRDTs and put it in .../crdt/CrdtInvariants.kt). It should be called something like invariant_mergeWithSelf_producesNoChanges.

File: javatests/arcs/core/crdt/CrdtEntityFuzzTest.kt [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import arcs.core.testutil.runFuzzTest
+import com.google.common.truth.Truth
+import org.junit.Test
+
+class CrdtEntityFuzzTest() {
+  /**
+   * Test that a randomly generated [CrdtEntity] merged with itself produces an empty [CrdtChange].
+   */
+  @Test
+  fun crdtEntity_merge_invariant() = runFuzzTest {
```
Comment (shans) [inline javatests/arcs/core/crdt/CrdtEntityFuzzTest.kt:17]: The test should be called something like mergeWithSelf_producesNoChanges (basically name it after the invariant but without the invariant_ prefix).

File: javatests/arcs/core/crdt/Generators.kt [hunk 4]
```diff
@@ -0,0 +1,65 @@
+package arcs.core.crdt
```
Comment (shans) [inline javatests/arcs/core/crdt/Generators.kt:1]: This file should be in java/arcs/core/crdt/testutil/

File: javatests/arcs/core/crdt/Generators.kt [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+      version(),
+      rawEntity()
+    )
+  }
+}
+
+/**
+ * Generate a [VersionMap] given generators for the actor and version.
+ */
+class VersionMapGenerator(
```
Comment (shans) [inline javatests/arcs/core/crdt/Generators.kt:25]: This only creates VersionMaps with a single actor, is that correct? Make this clear in the comment and by calling it SingleActorVersionMapGenerator

**Review by shans — APPROVED** [review]
This is looking great! Just a couple of naming / location comments.

File: javatests/arcs/core/crdt/CrdtEntityFuzzTest.kt (hunk 1 above)
Comment (SHeimlich) [reply javatests/arcs/core/crdt/CrdtEntityFuzzTest.kt]: Oops, it should be 3 separate options. I'm not sure how that happened....
